Python/Product/PythonTools/PythonTools/Project/PythonNotSupportedInfoBar.cs

```diff
@@ -0,0 +1,144 @@
+                new PythonVersionNotSupportedInfoBarInfo() {
+                    Action = action,
+                    Context = _infoBarData.Context,
+                    PythonVersion = _infoBarData.Configuration.Version
+                }
+            );
+        }
+
+        protected abstract Task SetPropertyAsync(string propertyName, string propertyValue);
+
+        protected class PythonSupportInfoBarData {
+            public InterpreterConfiguration Configuration { get; protected set; }
+            public bool IsInfoBarSuppressed { get; }
+            public string Context { get; }
+
+            public PythonSupportInfoBarData(InterpreterConfiguration configuration, bool isInfoBarSuppressed, string context) {
+                Configuration = configuration;
+                IsInfoBarSuppressed = isInfoBarSuppressed;
+                Context = context;
+            }
+        }
+    }
+
+    internal sealed class PythonNotSupportedProjectInfoBar : PythonNotSupportedInfoBar {
+        private PythonProjectNode _project;
+
+        public PythonNotSupportedProjectInfoBar(IServiceProvider site, PythonProjectNode projectNode) : base(site) {
+            _project = projectNode ?? throw new ArgumentNullException(nameof(projectNode));
+
+        }
+
+        public override async Task CheckAsync() {
+            PythonSupportInfoBarData infoBarData = new PythonSupportInfoBarData(
+                _project.ActiveInterpreter.Configuration,
+                _project.GetProjectProperty(_suppressInfoBarPropertyName).IsTrue(),
+                InfoBarContexts.Project
+                );
+
+            await InfoBarCreationRequest(infoBarData);
+        }
+
+        protected override Task SetPropertyAsync(string propertyName, string propertyValue) {
+            _project.SetProjectProperty(propertyName, propertyValue);
+            return Task.CompletedTask;
+        }
+    }
+
+    internal sealed class PythonNotSupportedWorkspaceInfoBar : PythonNotSupportedInfoBar {
+        private IPythonWorkspaceContext _workspaceContext;
+
+        public PythonNotSupportedWorkspaceInfoBar(IServiceProvider site, IPythonWorkspaceContext pythonWorkspaceContext) : base(site) {
+            _workspaceContext = pythonWorkspaceContext ?? throw new ArgumentNullException(nameof(pythonWorkspaceContext));
+        }
+
+        public override async Task CheckAsync() {
+            PythonSupportInfoBarData infoBarData = new PythonSupportInfoBarData(
+                _workspaceContext.CurrentFactory.Configuration,
+                _workspaceContext.GetBoolProperty(_suppressInfoBarPropertyName) ?? false,
+                InfoBarContexts.Workspace
+                );
+
+            await InfoBarCreationRequest(infoBarData);
+        }
+
+        protected override async Task SetPropertyAsync(string propertyName, string propertyValue) {
+            await _workspaceContext.SetPropertyAsync(propertyName, propertyValue);
```

Comment:
these one liners, you can skip the async and await, its more efficient to just return _workspaceContext.SetPropertyAsync